Source/Core/Core/DSP/DSPHWInterface.cpp

```diff
@@ -283,22 +282,15 @@ u16 gdsp_ifx_read(u16 addr)
 
 static const u8* gdsp_idma_in(u16 dsp_addr, u32 addr, u32 size)
 {
-  u16* dst = g_dsp.iram + (dsp_addr / 2);
-
-  const u8* code = &g_dsp.cpu_ram[addr & 0x0fffffff];
-  g_dsp.iram_crc = Common::HashEctor(code, size);
-
   Common::UnWriteProtectMemory(g_dsp.iram, DSP_IRAM_BYTE_SIZE, false);
-  memcpy(dst, code, size);
-  for (size_t i = 0; i < size / 2; i++)
-    dst[i] = Common::swap16(dst[i]);
+  Host::DMAToDSP(g_dsp.iram + dsp_addr / 2, addr, size);
```

Comment:
Is not the mask not needed anymore?

Reply:
Assuming you're referring to `g_dsp.cpu_ram[addr & 0x0fffffff]`, I don't think it is; `CopyFromEmuSwapped` uses [`Memory::GetPointer`](https://github.com/dolphin-emu/dolphin/blob/9ffa72ad1fb8574ecda9e120d8a7fb930d053b01/Source/Core/Core/HW/Memmap.cpp#L536-L553) and it does its own masking (though with a different mask).

Perhaps [`Memory::GetPointerForRange`](https://github.com/dolphin-emu/dolphin/blob/9ffa72ad1fb8574ecda9e120d8a7fb930d053b01/Source/Core/Core/HW/Memmap.cpp#L463-L475) would be better, but it's not public.